report_alternative_layout/report/report_template.xml

```diff
@@ -1,70 +1,142 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <odoo>
     <template id="external_layout_standard" inherit_id="web.external_layout_standard">
+        <!-- Hide the standard report_header  -->
+        <xpath expr="//div[@class='col-9 text-end']" position="attributes">
+            <attribute name="style">display: none;</attribute>
+        </xpath>
+        <!-- Display the header only if 'report_header' is set.
+        Handle auto-inserted <p><br></p> from HTML fields -->
+        <xpath expr="//div[@class='col-9 text-end']" position="after">
+            <div class="col-9 text-end">
+                <t
+                    t-if="company.report_header and company.report_header != '&lt;p&gt;&lt;br&gt;&lt;/p&gt;'"
+                >
+                    <span t-field="company.report_header" name="moto" />
+                </t>
+                <!-- Display the document number in header -->
+                <t t-if="report and report.show_document_number">
+                    <div class="col text-end" t-if="o.name">
+                        <strong>
+                            Document No:
+                            <span t-field="o.name" />
+                        </strong>
+                    </div>
+                </t>
+                <!-- Display the date field in header -->
+                <t t-if="report and report.date_field_id and o">
+                    <t t-set="date_value" t-value="report._get_date_field_value(o)" />
+                    <t t-if="date_value">
+                        <div class="col text-end">
+                            <t t-esc="report.date_field_id.field_description" />
```

Comment:
Non-blocking but it would be good to have a `date_field_label` field in the report and show its value if available, as we might want to make adjustment sometimes.